website/docs/build-apps/how-to-guides/insert-data.md

```diff
@@ -0,0 +1,55 @@
+---
+description: This page shows you how to insert data in Appsmith. 
+
+---
+# Insert Data
+
+This guide shows you how to insert data into a datasource using the Form widget in a new page and a Modal. It explains steps to bind data to widgets, validate user input, and configure query.
+
+## Using new page
+To insert data in a new page, follow these steps:
+1. Add a Button widget. Set the widget's **onClick** event to open a new page by selecting the **Navigate to** action and selecting the appropriate page.
+   For more information, see [navigateTo](/reference/appsmith-framework/widget-actions/navigate-to).
+2. In the new page, create a Form widget to capture the details you want to insert.
+3. You can update fields dynamically by setting the events of the widgets to bind data to other fields in the Form.
+   For example, to set the `offerCode` based on the selection of `location` for a new offer use the following code:
```

Comment:
Just write example. Below the example, give them context of what is the use of that code

Reply:
Done